lib/master.js

```diff
@@ -218,24 +235,25 @@ class Master extends BaseService {
                     worker_pid: worker.process.pid,
                     exit_code: worker.process.exitCode
                 };
-                if (this.workerStatusMap[worker.process.pid] &&
-                    this.workerStatusMap[worker.process.pid].status) {
-                    info.status = this.workerStatusMap[worker.process.pid].status;
+                if (this._workerStatusMap[worker.process.pid] &&
+                    this._workerStatusMap[worker.process.pid].status) {
+                    info.status = this._workerStatusMap[worker.process.pid].status;
                 }
                 this._logger.log('error/service-runner/master', info);
-                delete this.workerStatusMap[worker.process.pid];
+                delete this._workerStatusMap[worker.process.pid];
                 P.delay(Math.random() * 2000).then(() => {
                     resolve(this._startWorker(worker.worker_id));
                 });
             };
 
-            worker.on('exit', startupWorkerExit);
-            worker.on('message', (msg) => {
+            this._workerMessageHandler = (msg) => {
```

Comment:
It has to be a local variable.

Reply:
Same issue as with workerExit

Reply:
And same issue as with workerExit - we don't need to remove it from stopWorker, we need to remove it from workerExit.

I thought you said you understood my explanations on our meetings yesterday and it was clear how it all fits together. If it was not clear, why didn't you ask more followup questions?